civis/io/_tables.py

```diff
@@ -1027,10 +1027,22 @@ def civis_file_to_table(file_id, database, table, client=None,
     except ValueError:
         table_exists = False
 
+    table_columns_valid = True
+    if table_columns:
+        table_columns_valid = True if table_columns[0]['sql_type'] else False
+        for i in range(0, len(table_columns)-1):
+            this_sql_type = table_columns[i]['sql_type']
+            next_sql_type = table_columns[i+1]['sql_type']
+            if (not this_sql_type) == (next_sql_type):
```

Comment:
This was not actually working as expected, had to add parentheses to have them evaluate as needed.

Reply:
See John's comments about simplifying this approach, this takes a bit of effort to understand what exactly we are looking for. I like the idea of using the counts of how many `sql_type` are provided.